actionsystemtest/lib/action_system_test/server.rb

```diff
@@ -0,0 +1,34 @@
+require "rack/handler/puma"
+
+module ActionSystemTest
+  class Server # :nodoc:
+    def initialize(port)
+      @port = port
+    end
+
+    def run
+      register
+      setup
+    end
+
+    private
+      def register
+        Capybara.register_server :puma do |app, host|
+          Rack::Handler::Puma.run(app, Port: @port, Threads: "0:1")
```

Comment:
If this is meant to support testing with ActionCable in process I believe the maximum number of threads in the pool needs to be more than 1 -  It might make sense to just use the :puma server registration provided by Capybara rather than registering your own.